plugins/discovery-ec2/src/main/java/org/opensearch/discovery/ec2/AwsEc2ServiceImpl.java

```diff
@@ -99,7 +99,13 @@ protected Ec2Client buildClient(
 
         if (Strings.hasText(endpoint)) {
             logger.debug("using explicit ec2 endpoint [{}]", endpoint);
-            builder.endpointOverride(URI.create(endpoint));
+            URI uri = URI.create(endpoint);
```

Comment:
Can you add a unit test for this?  Thanks!

Reply:
Hi @kkmr, I made some attempts, Mockito cannot mock the Ec2Client.builder static method well.